src/actions/commands/actions.ts

```diff
@@ -3270,6 +3275,24 @@ class ActionJoinVisualMode extends BaseCommand {
   }
 }
 
+@RegisterAction
+class ActionJoinVisualBlockMode extends BaseCommand {
+  modes = [Mode.VisualBlock];
+  keys = ['J'];
+  mightChangeDocument = true;
+
+  public async exec(position: Position, vimState: VimState): Promise<VimState> {
+    let actionJoin = new ActionJoin();
+    const range = vimState.cursors[0];
+    const topLeft = visualBlockGetTopLeftPosition(range.start, range.stop);
+    const bottomRight = visualBlockGetBottomRightPosition(range.start, range.stop);
```

Comment:
Could you use `vimState.cursorStartPosition` and `vimState.cursorStopPosition` instead?